dso/model.py

```diff
@@ -24,13 +27,41 @@ class Datasource:
     CC_UTILS = 'cc-utils'
 
 
+def normalise_artefact_extra_id(
+    artefact_extra_id: dict[str, str],
+    artefact_version: str=None,
+) -> str:
+    '''
+    generate stable representation of `artefact_extra_id` and remove `version` key if
+    the specified version is identical to the given artefact version
```

Comment:
why special-handling for version? how about adding this to gci.componentmodel instead, like so:
-> id_tuple: returns (alphabetically) ordered tuple of (<id>, <value>)-pairs, which will always contain as many id-attrs as are available (always `name`, `version` if available, every attr from extra-id)?

this function's output might be used to also normalise into a str (we might use tuple's __str__ for that or invent a custom format, as you suggested). however, as separation character, I would suggest to use one that is forbidden within keys/values - did you check ocm-spec whether this is true for underscore?

Reply:
We added the special-handling for the version already a while ago ([ref](https://github.com/gardener/cc-utils/commit/ae7c03d76cfe7c3242829eb08400da033493639c)). The reason behind this was that some artefacts contained their version also in the extra identity field, however, not always stable across different OCI repositories. Because of that, the comparison of the extra identities failed sometimes, depending of which OCI repository was used. To allow a stable representation of the extra identity independent of the OCI repository, we decided to allow the removal of the version attribute from the extra identity if desired.

Also, I just moved the `normalise_artefact_extra_id` function from `delivery.model` to `dso.model`, so I did not look into it in more detail, e.g. the separating characters. Nevertheless, I figure the underscore character is very basic in Python variable names and thus not forbidden within keys/values.